msgpack4nim.nim

```diff
@@ -97,7 +97,7 @@ proc readStr*(s: MsgStream, length: int): string =
   result = newString(length)
   if length != 0:
     var L = s.readData(addr(result[0]), length)
-    if L != length: result.setLen(L)
+    doAssert(L == length, "string len mismatch")
```

Comment:
please raise exception, `IOError` for example. assertion error is to catch wrong logic in the library code or in the user code.

Reply:
Thanks for your review.
I changed to raise `IOError` instead of `AssertionError`.